src/hooks/useToggleAllChecked.ts

```diff
@@ -0,0 +1,33 @@
+import { useRecoilValue, useRecoilCallback } from 'recoil';
+import { useState } from 'react';
+import { productsIds } from '../store/selectors';
+import { isCheckedState } from '../store/atoms';
+import areAllItemsChecked from '../utils/areAllItemsChecked';
+import { CartItemType } from '../types';
+
+const useToggleAllChecked = ({ products }: { products: CartItemType[] }) => {
+  const productIds = useRecoilValue(productsIds);
+  const [allChecked, setAllChecked] = useState(areAllItemsChecked(productIds));
+
+  const handleToggleAll = useRecoilCallback(
+    ({ set }) =>
+      () => {
+        const newAllChecked = !allChecked;
+        setAllChecked(newAllChecked);
+
+        products.forEach((product) => {
+          set(isCheckedState(product.id), newAllChecked);
+          window.localStorage.setItem(JSON.stringify(product.id), JSON.stringify(newAllChecked));
+        });
```

Comment:
이전 상태를 참조할 때는 항상 주의해주세요..!

```suggestion
        setAllChecked(prev => {
          const newAllChecked = !prev;

          products.forEach((product) => {
            set(isCheckedState(product.id), newAllChecked);
            window.localStorage.setItem(JSON.stringify(product.id), JSON.stringify(newAllChecked));
          });
          
          return newAllChecked
        })
```

Reply:
앗 놓친 부분입니다! 짚어주셔서 감사합니다 🙇